crates/cxx-qt-lib/src/core/qtypes.rs

```diff
@@ -0,0 +1,83 @@
+// SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use cxx::{type_id, ExternType};
+use std::mem::MaybeUninit;
+
+#[cxx::bridge]
+mod ffi {
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/qtypes.h");
+
+        #[cxx_name = "qint64"]
+        type QInt64 = super::QInt64;
+
+        #[cxx_name = "qsizetype"]
+        type QSizeType = super::QSizeType;
+    }
+
+    #[namespace = "rust::cxxqtlib1"]
+    unsafe extern "C++" {
+        #[rust_name = "qint64_from_i64"]
+        fn qint64FromI64(value: i64) -> QInt64;
+        #[rust_name = "qint64_into_i64"]
+        fn qint64IntoI64(value: QInt64) -> i64;
+
+        #[rust_name = "qsizetype_from_isize"]
+        fn qsizetypeFromIsize(value: isize) -> QSizeType;
+        #[rust_name = "qsizetype_into_isize"]
+        fn qsizetypeIntoIsize(value: QSizeType) -> isize;
+    }
+}
+
+#[repr(C)]
+pub struct QInt64 {
```

Comment:
should probably have things like Eq, Ord, Debug ?

Or do we change this to be a totally trivial type?

Reply:
This then doesn't use `MaybeUninit`, do we need this as the memory might not be initialised and be random? But then it also is just a number :thinking:

Reply:
Gone with new type